scripts/autotools/libmongocrypt/Version.m4

```diff
@@ -0,0 +1,4 @@
+MONGOCRYPT_CURRENT_FILE=[]PHP_EXT_SRCDIR(mongodb)[/src/LIBMONGOCRYPT_VERSION_CURRENT]
+BUILD_VERSION=$(cat $MONGOCRYPT_CURRENT_FILE)
+
+AC_SUBST(BUILD_VERSION)
```

Comment:
Is there a particular reason this differs from the libmongoc and libbson `Versions.m4` file? Does libmongocrypt not have separate constants for version parts (e.g. major, minor)?

I'm a bit concerned that we're using a non-prefixed `BUILD_VERSION` constant here. I would expect libmongocrypt to at least use `MONGOCRYPT_VERSION` for the single version string.

@kevinAlbs: Is this something we can address in libmongocrypt, or has that ship sailed?

Reply:
If you're referring to renaming `BUILD_VERSION` to something more specific, that's fine with me. Though libmongoc also has the same cmake option to set a specific version to build with (https://github.com/mongodb/mongo-c-driver/blob/f930d74daf3fc9125e34330110a8510fd45c3ce5/CMakeLists.txt#L78)

Reply:
> Does libmongocrypt not have separate constants for version parts (e.g. major, minor)?

No, `MONGOCRYPT_VERSION` is the only version constant, containing the version as a string. This is comparable to the `MONGOC_VERSION_S` constant, but the other `MONGOC_*_VERSION` constants have no equivalent in libmongocrypt.

Reply:
@kevinAlbs My problem is with the `@BUILD_VERSION@` replacement in libmongocrypt's header file, since that's effectively within the same namespace as other replacements when we need to process the libmongoc and libbson files with Autotools. I think the CMake example you shared is setting a `BUILD_VERSION` _environment_ variable, since libmongoc does use prefixed replacement tokens in [`mongoc-version.h.in`](https://github.com/mongodb/mongo-c-driver/blob/master/src/libmongoc/src/mongoc/mongoc-version.h.in).

Assuming we're all on the same page: I don't think there's currently a conflict with other replacements (since libmongoc and libbson all prefix), but ideally I'd like to see this fixed alongside [MONGOCRYPT-218](https://jira.mongodb.org/browse/MONGOCRYPT-218) if you can create another ticket for it (likewise for @alcaeus to create a PHPC ticket as a todo reminder).

Reply:
The cmake example is referring to the same BUILD_VERSION cmake variable that is substituted in [mongocrypt.h.in](https://github.com/mongodb/libmongocrypt/blob/master/src/mongocrypt.h.in#L34), it is not an environment variable, unless I'm misunderstanding.

Created and scheduled https://jira.mongodb.org/browse/MONGOCRYPT-220

Reply:
I was thinking of the CMake environment when I wrote environment variable earlier. I was mainly differentiating its context from the replacement tokens in the `.in` file. Sorry for the confusion.